tests/testsuite/build.rs

```diff
@@ -5689,51 +5922,42 @@ fn build_filter_infer_profile() {
-        )
+        .with_stderr_data(str![[r#"
+[COMPILING] foo v0.0.1 ([ROOT]/foo)
+[RUNNING] `rustc --crate-name foo --edition=2015 src/lib.rs [..]--crate-type lib --emit=[..]link[..]`
+[RUNNING] `rustc --crate-name foo --edition=2015 src/main.rs [..]--crate-type bin --emit=[..]link[..]`
+[FINISHED] `dev` profile [unoptimized + debuginfo] target(s) in [ELAPSED]s
+
+"#]].unordered())
         .run();
 
     p.root().join("target").rm_rf();
     p.cargo("build -v --test=t1")
-        .with_stderr_contains(
-            "[RUNNING] `rustc --crate-name foo --edition=2015 src/lib.rs [..]--crate-type lib \
-             --emit=[..]link[..]-C debuginfo=2 [..]",
-        )
-        .with_stderr_contains(
-            "[RUNNING] `rustc --crate-name t1 --edition=2015 tests/t1.rs [..]--emit=[..]link[..]\
-             -C debuginfo=2 [..]",
-        )
-        .with_stderr_contains(
-            "[RUNNING] `rustc --crate-name foo --edition=2015 src/main.rs [..]--crate-type bin \
-             --emit=[..]link[..]-C debuginfo=2 [..]",
-        )
+        .with_stderr_data(str![[r#"
+[COMPILING] foo v0.0.1 ([ROOT]/foo)
+[RUNNING] `rustc --crate-name foo --edition=2015 src/lib.rs [..]--crate-type lib --emit=[..]link[..] -C debuginfo=2 [..]`
+[RUNNING] `rustc --crate-name foo --edition=2015 src/main.rs [..]--crate-type bin --emit=[..]link[..] -C debuginfo=2 [..]`
+[RUNNING] `rustc --crate-name t1 --edition=2015 tests/t1.rs [..]--emit=[..]link[..] -C debuginfo=2 [..]`
+[FINISHED] `dev` profile [unoptimized + debuginfo] target(s) in [ELAPSED]s
+
+"#]].unordered())
         .run();
 
     p.root().join("target").rm_rf();
     // Bench uses test profile without `--release`.
     p.cargo("build -v --bench=b1")
-        .with_stderr_contains(
-            "[RUNNING] `rustc --crate-name foo --edition=2015 src/lib.rs [..]--crate-type lib \
-             --emit=[..]link[..]-C debuginfo=2 [..]",
-        )
-        .with_stderr_contains(
-            "[RUNNING] `rustc --crate-name b1 --edition=2015 benches/b1.rs [..]--emit=[..]link[..]\
-             -C debuginfo=2 [..]",
-        )
-        .with_stderr_does_not_contain("opt-level")
-        .with_stderr_contains(
-            "[RUNNING] `rustc --crate-name foo --edition=2015 src/main.rs [..]--crate-type bin \
-             --emit=[..]link[..]-C debuginfo=2 [..]",
-        )
+        .with_stderr_data(str![[r#"
+[COMPILING] foo v0.0.1 ([ROOT]/foo)
+[RUNNING] `rustc --crate-name foo --edition=2015 src/lib.rs [..]--crate-type lib --emit=[..]link [..]-C debuginfo=2 [..]`
+[RUNNING] `rustc --crate-name b1 --edition=2015 benches/b1.rs [..]--emit=[..]link[..] -C embed-bitcode=no -C debuginfo=2 [..]--test [..]`
+[RUNNING] `rustc --crate-name foo --edition=2015 src/main.rs [..]--crate-type bin --emit=[..]link [..]-C debuginfo=2 [..]`
+[FINISHED] `dev` profile [unoptimized + debuginfo] target(s) in [ELAPSED]s
+
+"#]].unordered())
         .run();
 }
 
+#[allow(deprecated)]
 #[cargo_test]
 fn targets_selected_default() {
```

Comment:
This is an example of a test that uses `does_not_contain` in a way that I don't trust.  It is hard coding specific rust flags and assuming from the lack of that that different targets aren't being built.

Unsure what to do about it though

Reply:
A non-auto-updatable snapshot would help? So that people won't update it accidentally.